File: CppPractice/CppLeetcode/Leetcode415.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        reverse(num1.begin(), num1.end());
        reverse(num2.begin(), num2.end());
        int ptr1=0;
        int ptr2=0;
        int carry = 0;
        string ans ="";
        while (ptr1<num1.size() || ptr2<num2.size1)
        {
            int d1=0;
            int d2=0;
            if (ptr1<num1.size())
            {
                d1 = num1[ptr1]-'0';
                ptr1++;
            }
            if (ptr2<num2.size())
            {
                d2=nums[ptr2]-'0';
                ptr2++;
            }
            ans += to_string((d1+d2+carry)%10);
            carry = (d1+d2+carry)/10;
        }
        if (caryy>0)
        {
            ans+=to_string(carry);
        }
        reverse(ans.begin(), ans.end());
        return ans;
    
    }   
};

File: CSharpPractice/LeetcodePractice/Leetcode415.cs
public class Solution415 {
    public string AddStrings(string num1, string num2) {
        string ans="";
        int ptr1 = num1.Length-1;
        int ptr2 = num2.Length-1;
        int carry=0;
        while (ptr1>=0||ptr2>=0||carry>0)
        {
            int d1=0;
            int d2=0;
            if (ptr1>=0)
            {
                d1 = num1[ptr1--]-'0';   
            }
            if (ptr2>=0)
            {
                d2 = num2[ptr2--]-'0';   
            }
            ans += ((d1+d2+carry)%10).ToString();
            carry = (d1+d2+carry)/10;
        }
        string reverseAns = new string(ans.Reverse().ToArray());
        return reverseAns;
    }
}